posthog/session_recordings/session_recording_api.py

```diff
@@ -835,27 +835,32 @@ def _gather_session_recording_sources(
 
         with GATHER_RECORDING_SOURCES_HISTOGRAM.labels(blob_version="v2" if is_v2_enabled else "v1").time():
             if is_v2_enabled:
-                with timer("list_blocks__gather_session_recording_sources"):
-                    blocks = list_blocks(recording)
-
-                for i, block in enumerate(blocks):
-                    sources.append(
-                        {
-                            "source": "blob_v2",
-                            "start_timestamp": block.start_time,
-                            "end_timestamp": block.end_time,
-                            "blob_key": str(i),
-                        }
-                    )
-
-            with timer("list_objects__gather_session_recording_sources"):
-                if recording.object_storage_path:
-                    blob_prefix = recording.object_storage_path
+                if recording.full_recording_v2_path:
+                    blob_prefix = recording.full_recording_v2_path
```

Comment:
question: Did you test that it works? The way S3 permissions are handled in production in V2 is completely different, so I'm not sure if this is gonna work.

Reply:
Yeah, I don't exactly know how it could break, my feeling is that the S3 bucket is pretty locked up and the clients won't be able to access the files. I'm also a bit wary of exposing S3 paths to the clients, tbh.

Happy for you to make the decision though, as you own the API. Releasing it behind a feature flag would be great.